### PR TITLE
Fix unmatch colors when props in IChartData is empty

### PR DESCRIPTION
**Pull request by ervfreitas**
## Previous Behavior

The colors in Donut and Legends don't match
## New Behavior
Show the same palette color for item in Donut and Legends

## Related Issue(s)

https://github.com/microsoft/fluentui/issues/29018

- Change logic to add default paletter colors for empty dataChart props

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/1cc078ada94cfe929c0aa8d60ea100cea370795d'>1cc078ada94cfe929c0aa8d60ea100cea370795d</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: change/@fluentui-react-charting-ebb094dd-b52c-43f1-b604-00481d4733c0.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-charting-ebb094dd-b52c-43f1-b604-00481d4733c0.json:1]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=29113&commitId=551fc1b95c4249488c2ced69bcc795acfc893010&env=prod&clientType=FLUENTUI)


  <details><summary><b>ActivityItem</b> 4 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|ActivityItem.Personas - RTL.default.chromium.png|0| Added|
|ActivityItem.Personas.default.chromium.png|0| Added|
|ActivityItem.Root - RTL.default.chromium.png|0| Added|
|ActivityItem.Root.default.chromium.png|0| Added|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

Comment (AtishayMsft) [reply change/@fluentui-react-charting-ebb094dd-b52c-43f1-b604-00481d4733c0.json:1]: Screenshot change not related to this PR.

**Review by AtishayMsft — APPROVED** [review]

**Comment by AtishayMsft** [comment]
> @ervfreitas please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@ervfreitas Could you agree on the contributor license so that I can merge these changes.

**Comment by ervfreitas** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Comment by ervfreitas** [comment]
> > @ervfreitas please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> > ```
> > @microsoft-github-policy-service agree [company="{your company}"]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > > Options:
> > > 
> > > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > > 
> > > ```
> > > @microsoft-github-policy-service agree
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 
> > > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > > 
> > > ```
> > > @microsoft-github-policy-service agree company="Microsoft"
> > > ```
> > 
> > 
> > Contributor License Agreement
> 
> @ervfreitas Could you agree on the contributor license so that I can merge these changes.

@microsoft-github-policy-service agree company="Microsoft"

**Comment by AtishayMsft** [comment]
Thanks @ervfreitas for contributing the bug fix and help improve the react-charting library.

**Pull request merged**